src/renderer/components/composer/ComposerMessageInput.tsx

```diff
@@ -28,14 +30,44 @@ export default class ComposerMessageInput extends React.Component<
   setCursorPosition: number | false
   textareaRef: React.RefObject<HTMLTextAreaElement>
   saveDraft: () => void
+  recorder: MediaRecorder | null = null
+  updateRecordedDurationInterval: number | null = null
+  voiceData: Blob[] = []
+  gumError: any
   constructor(props: ComposerMessageInputProps) {
     super(props)
     this.state = {
       text: '',
       chatId: props.chatId,
       loadingDraft: true,
+      recordedDuration: null,
     }
-
+    navigator.mediaDevices.getUserMedia({ audio: true, video: false }).then(stream => {
+      this.recorder = new MediaRecorder(stream)
+      this.recorder.onstart = () => {
+        this.updateRecordedDurationInterval = window.setInterval(() => 
+          this.setState(prevState => {
+            let recordedDuration = prevState.recordedDuration?.clone() || moment.duration()
+            recordedDuration.add(0.1, 's')
+            return { recordedDuration: recordedDuration }
+          })
+        , 100)
+      }
+      this.recorder.ondataavailable = (evt: BlobEvent) => this.voiceData.push(evt.data)
+      this.recorder.onstop = () => {
+        let duration = this.state.recordedDuration?.asSeconds() || 0
+        if (duration === 0)
+          log.error('duration of voice is zero while the mediarecorder has stopped. this must not happen')
+        this.setState({ recordedDuration: null })
+        // send the voice message
+        console.log(this.voiceData)
+        this.voiceData = []
+        this.updateRecordedDurationInterval && window.clearInterval(this.updateRecordedDurationInterval)
+      }
+    })
+    .catch((reason: any) => {
+      this.gumError = reason
```

Comment:
please also log this error with `log.error`, maybe we should even somehow show it to the user later, maybe in an alert dialog?